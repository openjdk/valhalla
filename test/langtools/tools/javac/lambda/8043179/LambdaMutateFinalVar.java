/*
 * @test /nodynamiccopyright/
 * @summary Verify lambda expression can't mutate a final variable
 * @bug 8043179
 * @compile/fail/ref=LambdaMutateFinalVar.out -XDrawDiagnostics LambdaMutateFinalVar.java
 */
class LambdaMutateFinalVar {
    LambdaMutateFinalVar() {
        final String x;
        Runnable r1 = () -> x = "not ok";
        x = "ok";
    }
}
