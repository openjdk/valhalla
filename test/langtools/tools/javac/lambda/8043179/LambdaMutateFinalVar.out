LambdaMutateFinalVar.java:10:29: compiler.err.var.might.already.be.assigned: x
1 error
